chainerx_cc/chainerx/routines/manipulation.h

```diff
@@ -22,6 +28,11 @@ Array RollAxis(const Array& a, int8_t axis, int8_t start = 0);
 // Returns a transposed view of the array.
 Array Transpose(const Array& a, const OptionalAxes& axes = nonstd::nullopt);
 
+// Returns an array padded by pad_width amount
+Array Pad(const Array& a, int8_t pad_width, PadMode mode, int64_t constant_values);
```

Comment:
The types of `pad_width` and `constant_values` are independent.
For example, one should be able to pass `int8_t` and `std::vector<std::vector<int64_t>>`.
One way would be to introduce specialized classes `class PadWidths` and `class ConstantValues` which are implicitly convertible from `int8_t`/`std::vector<int8_t>`/`std::vector<std::vector<int8_t>>` (replace `int8_t` with `int64_t` for `ConstantValues`).

Reply:
Could you please elaborate a bit how to implement this? I have been stuck with this since past few days.